README.rst

```diff
@@ -207,23 +207,59 @@ And decrypt it using::
 
 	 $ sops --decrypt test.enc.yaml
 
+Encrypting using Azure Key Vault
+~~~~~~~~~~~~~~~~~~~~~~~~
+The Azure Key Vault integration uses service principals to access secrets in
+the vault. The following environment variables are used to authenticate:
+
+.. code:: bash
+
+	AZURE_TENANT_ID
```

Comment:
Where do I get these from?

I understand you can generate "identities" for applications, but most people will want to operate as their user. How do they do that?

Reply:
Yeah, I'd like that too... But from my understanding of the Azure SDK, you always need a client for your application. For example, the `az` CLI creates a new application identity/service principal for itself. 
The AZURE_USERNAME and AZURE_PASSWORD env vars are read, but you still need to provide a client id, and I haven't been able to figure out which client id works with those.

So as far as I know, you cannot use your normal user/password :disappointed:
Not all that deep into their SDKs yet, though, so I may be missing something. 

I'll add some docs on how to generate an SP, though.

Reply:
Ok, so there is something called a device flow, which can also be used (and which is what the az cli does to generate its SP). Perhaps that might be a good thing for interactive use

Reply:
A link to the Azure docs on what goes into each env var and how to generate each thing would be ideal. What's important is that users who aren't familiar with Azure (such as myself) have the basics to know how to use SOPS with it. For example, someone who's been using Azure for years surely knows how to get local tools to talk to Azure services, but I have no idea. I expected it to just work after doing `az login`. For example, this is why we explain you need your AWS credentials in `~/.aws` in the README. 

In summary, we need users who might be new to using third party CLI tools with their cloud provider to be able to use SOPS just by reading our docs (and the docs we link to from our docs).

Does this make sense?

I agree that we don't need the `device flow` on this PR. I just want users to be able to use Azure by reading this docs somehow.

Reply:
Agree in full! Did you see the new commit I pushed? It adds the cli invokations for creating a new principal, and how the output maps to the environment variables. I'll see if I can find some extra official docs to link to. They tend to not be quite focused on where to click in the portal, though...

Reply:
Oh, hah! You pushed this after I started writing this and the page didn't update. It's perfect now. Thanks!